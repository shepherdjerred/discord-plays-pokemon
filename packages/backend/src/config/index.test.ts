import { describe, it, expect } from "vitest";
import { getConfig } from "./index.js";

describe("config", () => {
  it("should not accept the default configuration", () => {
    try {
      getConfig("config.example.toml");
    } catch (e) {
      expect(e).toMatchSnapshot("the configuration should require non-empty fields");
    }
  });
});
