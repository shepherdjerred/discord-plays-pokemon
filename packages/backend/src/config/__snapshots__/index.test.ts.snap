// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`config > should not accept the default configuration > the configuration should require non-empty fields 1`] = `
[Error: Your configuration at /Users/jerred/git/discord-plays-pokemon/config.example.toml _is_ valid TOML, but it is not a valid configuration for this application.

Code: too_small ~ Path: server_id ~ Message: String must contain at least 1 character(s)
Code: too_small ~ Path: bot.discord_token ~ Message: String must contain at least 1 character(s)
Code: too_small ~ Path: bot.application_id ~ Message: String must contain at least 1 character(s)
Code: too_small ~ Path: bot.notifications.channel_id ~ Message: String must contain at least 1 character(s)
Code: too_small ~ Path: stream.channel_id ~ Message: String must contain at least 1 character(s)
Code: too_small ~ Path: stream.userbot.id ~ Message: String must contain at least 1 character(s)
Code: too_small ~ Path: stream.userbot.username ~ Message: String must contain at least 1 character(s)
Code: too_small ~ Path: stream.userbot.password ~ Message: String must contain at least 1 character(s)
Code: too_small ~ Path: game.commands.channel_id ~ Message: String must contain at least 1 character(s)

For more information, read the setup guide (https://docs.discord-plays-pokemon.com/user/)
If you are unable to resolve this error, please open an issue (https://github.com/shepherdjerred/discord-plays-pokemon/issues)
]
`;
